tests/v2prov/tests/fleet/fleetcluster_test.go

```diff
@@ -0,0 +1,134 @@
+				{
+					Weight: 1,
+					Preference: corev1.NodeSelectorTerm{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "fleet.cattle.io/agent",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	linuxAffinity = corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{
+					{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "kubernetes.io/os",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"linux"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	resourceReq = &corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU: resource.MustParse("1"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse("1Gi"),
+		},
+	}
+	tolerations = []corev1.Toleration{
+		{
+			Key:      "key",
+			Operator: corev1.TolerationOpEqual,
+			Value:    "value",
+		},
+	}
+)
+
+func TestFleetCluster(t *testing.T) {
+	require := require.New(t)
+	clients, err := clients.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer clients.Close()
+
+	lc := &fleetv1api.Cluster{}
+	// wait for fleet local cluster with default affinity
+	require.Eventually(func() bool {
+		lc, err = clients.Fleet.Cluster().Get("fleet-local", "local", metav1.GetOptions{})
+		return err == nil && lc != nil && lc.Status.Summary.Ready > 0
+	}, waitFor, tick)
+	require.Equal(lc.Spec.AgentAffinity, &builtinAffinity)
+	require.Nil(lc.Spec.AgentResources)
+	require.Empty(lc.Spec.AgentTolerations)
+
+	// fleet-agent deployment has affinity
+	agent, err := clients.Apps.Deployment().Get(lc.Status.Agent.Namespace, "fleet-agent", metav1.GetOptions{})
+	require.NoError(err)
+	require.Equal(agent.Spec.Template.Spec.Affinity, &builtinAffinity)
+	require.Len(agent.Spec.Template.Spec.Containers, 1)
+	require.Empty(agent.Spec.Template.Spec.Containers[0].Resources)
+	require.NotEmpty(agent.Spec.Template.Spec.Tolerations) // Fleet has built-in tolerations
+
+	// change settings on management cluster, results should show up in fleet-agent deployment
+	mc, err := clients.Mgmt.Cluster().Get("local", metav1.GetOptions{})
+	require.NoError(err)
+
+	mc.Spec.FleetAgentDeploymentCustomization = &mgmt.AgentDeploymentCustomization{
+		OverrideAffinity:             &linuxAffinity,
+		OverrideResourceRequirements: resourceReq,
+		AppendTolerations:            tolerations,
+	}
+
+	_, err = clients.Mgmt.Cluster().Update(mc)
+	require.NoError(err)
+
+	// changes propagate to fleet cluster
+	require.Eventually(func() bool {
+		lc, err = clients.Fleet.Cluster().Get("fleet-local", "local", metav1.GetOptions{})
+		return err == nil && lc != nil && lc.Status.Summary.Ready > 0 && assert.Equal(t, lc.Spec.AgentAffinity, &linuxAffinity)
```

Comment:
Ah, I see the problem with the complexity. Not sure if this is much better, though since both the explicit false and the false case from assert will trigger a retry of the out `Eventually` 🤷 
```
		if err == nil && cluster != nil && cluster.Status.Summary.Ready > 0 {
			assert.Equal(t, cluster.Spec.AgentAffinity, &linuxAffinity)
		}
		return false
```